src/contracts/ChiToken.sol

```diff
@@ -119,102 +119,130 @@ contract ChiToken is IERC20, ERC20WithoutTotalSupply {
         return totalMinted - totalBurned;
     }
 
-    function mint(uint256 value) public {
-        uint256 offset = totalMinted;
+    // Writes the 36 bytes of subcontract creation code to memory at the specified offset
+    function write_bytecode_to_memory(uint256 offset) internal view {
+        // Documentation adapted from https://github.com/projectchicago/gastoken/blob/master/contract/GST2_ETH.sol#L105
+        //
+        // EVM assembler of runtime portion of child contract:
+        //     ;; Pseudocode: if (msg.sender != <address>) { throw; }
+        //     ;;             suicide(msg.sender)
+        //     PUSH20 <address>
+        //     CALLER
+        //     XOR
+        //     PC
+        //     JUMPI
+        //     CALLER
+        //     SELFDESTRUCT
+        // Or in binary: 73____________20 bytes address____________3318585733ff
+        // Since the binary is so short (27 bytes), we can get away
+        // with a very simple initcode:
+        //     PUSH27 73____________20 bytes address____________3318585733ff
+        //     PUSH1 0
+        //     MSTORE ;; at this point, memory locations mem[5] through
+        //            ;; mem[31] contain the runtime portion of the child
+        //            ;; contract. all that's left to do is to RETURN this
+        //            ;; chunk of memory.
+        //     PUSH1 27 ;; length
+        //     PUSH1 5 ;; offset
+        //     RETURN
+        // Or in binary: 7a73____________20 bytes address____________3318585733ff600052601b6005f3
+        // Almost done! All we have to do is put this short (36 bytes) blob into
+        // memory and call CREATE with the appropriate offsets.
         assembly {
             mstore(
-                0,
-                0x746d4946c0e9F43F4Dee607b0eF1fA1c3318585733ff6000526015600bf30000
+                offset,
+                add(
+                    add(
+                        0x7a73000000000000000000000000000000000000000000000000000000000000,
+                        shl(0x50, address())
+                    ),
+                    0x3318585733ff60005260
```

Comment:
ubernit: To me this would have been slightly clearer if the `0`s were the space for the address:
```suggestion
                add(
                    or(
                        0x7a7300000000000000000000000000000000000000003318585733ff60005260,
                        shl(0x50, address())
                    ),
```